fiftyone/factory/repos/execution_store.py

```diff
@@ -198,21 +198,42 @@ def has_store_global(self, store_name):
         )
         return bool(result)
 
-    def list_stores_global(self) -> list[str]:
-        """Lists the stores in the execution store across all datasets and the
-        global context.
-        """
-        result = self._collection.find(
-            dict(key="__store__"), {"store_name": 1}
-        )
-        return [d["store_name"] for d in result]
+    def list_stores_global(self) -> list[StoreDocument]:
+        """Lists stores across all datasets and the global context."""
+        pipeline = [
+            {
+                "$group": {
+                    "_id": {
+                        "store_name": "$store_name",
+                        "dataset_id": "$dataset_id",
+                    }
+                }
+            },
+            {
+                "$project": {
```

Comment:
The `key=__store__` document is optional. This was returning no results in a few of the integration tests.

Another option would be to ensure a store document existed in set_key, which is an extra mongo operation.

Reply:
Ah okay good catch 👍